### PR TITLE
Use dstr() in dm strings for thinpool and thin devs

### LINKED ISSUE #19: Consistency between lineardev and thindev's use of dstr
@mulkieran points out in #13 

Why does lineardev use the device's dstr in the dmtable, but for thindev and thinpooldev the device node is used instead?

### PR DESCRIPTION
**Pull request by agrover**
Add dstr() methods for linear, thinpool, and thin devs.

Use dstr() in dm_table methods for thinpooldev and thindev.

fixes #19

Signed-off-by: Andy Grover <agrover@redhat.com>

### CODE REVIEW

File: src/thindev.rs [hunk 1]
```diff
@@ -38,10 +36,9 @@ impl ThinDev {
                length: Sectors)
                -> DmResult<ThinDev> {
 
-        let thin_dev_path = format!("{}", try!(thin_pool.devnode()).to_string_lossy());
         try!(thin_pool.message(dm, &format!("create_thin {}", thin_id)));
         try!(dm.device_create(name, None, DmFlags::empty()));
-        let table = ThinDev::dm_table(&thin_dev_path, thin_id, length);
+        let table = ThinDev::dm_table(&thin_pool, thin_id, length);
```
Comment (mulkieran) [inline src/thindev.rs]: ```table``` is only used in one place. Maybe you could substitute the ```ThinDev``` expression at that location.

File: src/thindev.rs [hunk 2]
```diff
@@ -55,11 +52,11 @@ impl ThinDev {
 
     /// Generate a Vec<> to be passed to DM.  The format of the Vec entries are:
     /// "<start> <length> thin </dev/mapper/poolname> <thin_id>"
-    fn dm_table(pool_dev_str: &String,
+    fn dm_table(thin_pool: &ThinPoolDev,
                 thin_id: u32,
                 length: Sectors)
                 -> Vec<(u64, u64, String, String)> {
```
Comment (mulkieran) [inline src/thindev.rs]: Aside: e.g., return type here is really TargetLine.

File: src/thinpooldev.rs [hunk 3]
```diff
@@ -44,13 +43,7 @@ impl ThinPoolDev {
...
         try!(dm.device_create(name, None, DmFlags::empty()));
 
-        let meta_dev_path = try!(meta.devnode());
-        let data_dev_path = try!(data.devnode());
-        let table = ThinPoolDev::dm_table(length,
-                                          data_block_size,
-                                          low_water_mark,
-                                          &meta_dev_path,
-                                          &data_dev_path);
+        let table = ThinPoolDev::dm_table(length, data_block_size, low_water_mark, &meta, &data);
```
Comment (mulkieran) [inline src/thinpooldev.rs]: Substituting expression for name would work here as well.

File: src/thinpooldev.rs [hunk 4]
```diff
@@ -68,12 +61,12 @@ impl ThinPoolDev {
     fn dm_table(length: Sectors,
```
Comment (mulkieran) [inline src/thinpooldev.rs]: Header comment for dm_table needs to be updated.

File: src/thindev.rs [hunk 5]
```diff
@@ -55,11 +52,11 @@ impl ThinDev {
 
     /// Generate a Vec<> to be passed to DM.  The format of the Vec entries are:
     /// "<start> <length> thin </dev/mapper/poolname> <thin_id>"
```
Comment (mulkieran) [inline src/thindev.rs]: Fix header comment.

File: src/thinpooldev.rs [hunk 6]
```diff
@@ -68,12 +61,12 @@ impl ThinPoolDev {
...
                 data_block_size: Sectors,
                 low_water_mark: DataBlocks,
-                meta_dev: &Path,
-                data_dev: &Path)
+                meta: &LinearDev,
+                data: &LinearDev)
                 -> Vec<(u64, u64, String, String)> {
         let params = format!("{} {} {} {} 1 skip_block_zeroing",
```
Comment (mulkieran) [inline src/thinpooldev.rs]: TargetLine return type desirable here.

**Comment by trgill** [comment]
👍

**Pull request merged**